pyro/infer/advi.py

```diff
@@ -43,9 +43,10 @@ def _setup_prototype(self, *args, **kwargs):
         # run the model so we can inspect its structure
         self.prototype_trace = poutine.block(poutine.trace(self.base_model).get_trace)(*args, **kwargs)
         self.prototype_trace = prune_subsample_sites(self.prototype_trace)
-        self.latent_dim = sum(site["value"].view(-1).size(0)
-                              for site in self.prototype_trace.nodes.values()
-                              if site["type"] == "sample" and not site["is_observed"])
+        self.latent_shapes = {name: biject_to(site["fn"].support).inv(site["value"]).shape
```

Comment:
why is this necessary? can we have a comment?

Reply:
comment added.